packages/cw721/src/msg.rs

```diff
@@ -201,9 +288,513 @@ pub enum Cw721MigrateMsg {
+{
+    fn create(
+        &self,
+        deps: Option<Deps>,
+        env: Option<&Env>,
+        info: Option<&MessageInfo>,
+        current: Option<&CollectionInfoAndExtensionResponse<TCollectionExtension>>,
+    ) -> Result<CollectionInfoAndExtensionResponse<TCollectionExtension>, Cw721ContractError> {
+        self.validate(deps, env, info, current)?;
+        match current {
+            // Some: update existing metadata
+            Some(current) => {
+                let mut updated = current.clone();
+                if let Some(name) = &self.name {
+                    updated.name = name.clone();
+                }
+                if let Some(symbol) = &self.symbol {
+                    updated.symbol = symbol.clone();
+                }
+                let current_extension = current.extension.clone();
+                let updated_extension =
+                    self.extension
+                        .create(deps, env, info, Some(&current_extension))?;
+                updated.extension = updated_extension;
+                Ok(updated)
+            }
+            // None: create new metadata
+            None => {
+                let extension = self.extension.create(deps, env, info, None)?;
+                let env = env.ok_or(Cw721ContractError::NoEnv)?;
+                let new = CollectionInfoAndExtensionResponse {
+                    name: self.name.clone().unwrap(),
+                    symbol: self.symbol.clone().unwrap(),
+                    extension,
+                    updated_at: env.block.time,
+                };
+                Ok(new)
+            }
+        }
+    }
+
+    fn validate(
+        &self,
+        deps: Option<Deps>,
+        _env: Option<&Env>,
+        info: Option<&MessageInfo>,
+        _current: Option<&CollectionInfoAndExtensionResponse<TCollectionExtension>>,
+    ) -> Result<(), Cw721ContractError> {
+        // make sure the name and symbol are not empty
+        if self.name.is_some() && self.name.clone().unwrap().is_empty() {
+            return Err(Cw721ContractError::CollectionNameEmpty {});
+        }
+        if self.symbol.is_some() && self.symbol.clone().unwrap().is_empty() {
+            return Err(Cw721ContractError::CollectionSymbolEmpty {});
+        }
+        let deps = deps.ok_or(Cw721ContractError::NoDeps)?;
+        // collection metadata can only be updated by the creator. creator assertion is skipped for these cases:
+        // - CREATOR store is empty/not initioized (like in instantiation)
+        // - info is none (like in migration)
+        let creator_initialized = CREATOR.item.may_load(deps.storage)?;
+        if (self.name.is_some() || self.symbol.is_some())
+            && creator_initialized.is_some()
+            && info.is_some()
+            && CREATOR
+                .assert_owner(deps.storage, &info.unwrap().sender)
+                .is_err()
+        {
+            return Err(Cw721ContractError::NotCreator {});
+        }
+        Ok(())
+    }
+}
```

Comment:
Finally here's the validation of collection info, like:
- name and symbol must not be empty
- sender must be creator